### DISCUSSION TITLE
Get API Routes know about current locale

### DISCUSSION THREAD

**Discussion by abalmus** [vercel/next.js#29227] [Help]
Hi guys, hope I find you well and healthy. 

I'm stuck with some simple task. I can't let my GraphQL API route know the current site locale. In case of pages we use  `getServerSideProps` which has a locale in props. 

Say I have a nextjs website `domain.com/en` which makes a request to the API route: `domain.com/api/graphql`. This is just a simple Apollo server. So at this stage I have to get a current locale which is `en` . Is it possible? Is there any way except setting some domain level cookie? 

```
export default function handler(req, res) {
  // get locale here 
}
```

**Comment by delbaoliveira** [comment]
Hi @abalmus! 👋🏼 

By default, Next.js will automatically try to detect the locale based on the user's preference ([more information on this](router)). 

You can access the locale information using two methods: 

1. With the **`useRouter` hook**
2. With the **context**  inside `getStaticProps()` or `getServerSideProps()`

When using the Apollo Server and `getServerSideProps()`, you will need to write a resolver to use context. We have an [example here ](https://github.com/vercel/next.js/blob/canary/examples/api-routes-graphql/pages/api/graphql.js) and I recommend looking at the Apollo docs on [resolvers](https://www.apollographql.com/docs/apollo-server/data/resolvers/).

Here's an alternative solution, also using a resolver, with `getServerSideProps()`: https://github.com/vercel/next.js/discussions/18250

**Reply by abalmus** [reply]
Thank for answering. The methods you described are working with Pages, not sure I can use them for API routes.

**Reply by abalmus** [reply]
Also the link " (more information on this)." doesn't work for me
